torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+"""Northeastern China Crop Map Dataset."""
+
+import glob
+import os
+import shutil
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NCCM(RasterDataset):
+    """The Northeastern China Crop Map Dataset.
+
+    Link: https://www.nature.com/articles/s41597-021-00827-9
+
+    This dataset produced annual 10-m crop maps of the
+    major crops (maize, soybean, and rice)
+    in Northeast China from 2017 to 2019, using hierarchial mapping strategies,
+    random forest classifiers, interpolated and
+    smoothed 10-day Sentinel-2 time series data and
+    optimized features from spectral, temporal and
+    textural characteristics of the land surface.
+    The resultant maps have high overall accuracies (OA)
+    based on ground truth data. The dataset contains information
+    specific to three years: 2017, 2018, 2019.
+
+    The dataset contains 4 classes:
+
+    0. paddy rice
+    1. maize
+    2. soybean
+    3. others
+
+    Dataset format:
+
+    1. Three .TIF files containing the labels
+    2. JavaScript code to download images from the dataset.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1038/s41597-021-00827-9
+    """
+
+    filename_regex = r"CDL(?P<year>\d{4})_clip"
+    filename_glob = "CDL*.tif"
```

Comment:
```suggestion
    filename_glob = "CDL*.*"
```
Then it will still work if someone converts the files to a different type